pkgs/servers/mobilizon/default.nix

```diff
@@ -7,17 +7,17 @@
 , git
 , cmake
 , nixosTests
+, srcOverride ? null
```

Comment:
I'm not sure I understand the usecase for this variable. This isn't a pattern I've seen in other packages. If one wants to change the source for the mobilizon package, shouldn't it be done like this:?

```nix
{
  nixpkgs.overlays = (final: prev: {
    mobilizon = prev.mobilizon.overrideAttrs (old: { src = ...; });
  });
}
```

But I might be missing something

Reply:
Yeah, that would be a nice way to do it, but unfortunately `overrideAttrs` doesn't alter the `src` of the `js.nix`, so the two derivations will use different sources. So you could only change elixir related stuff this way.

The only other application of the pattern I know of is the mastodon package, which suffers from the same problem.

If you know of any way to pass the _overriden_ src to other modules, I'd gladly implement it.

Reply:
Wouldn't be better to have ` { mobilizonJavaScript, ... }` in the head of `default.nix`, then, when we do `callPackage ./../mobilizon { mobilizonJavaScript = callPackage ../../mobilizon/js.nix { }; };`

so that JS can be overridable if needed at all.

Reply:
@minijackson Mh I did it exactly as in your snippet? Via nixpkgs.overlays https://github.com/NixOS/nixpkgs/pull/119132#discussion_r847241203

@RaitoBezarius Couldn't we then make mobilizon-frontend a top-level attribute? This way you can override the src even easier.

Reply:
@erictapen we could definitely I guess, or we could create a root package set behind `mobilizon` such that `mobilizon.frontend` exist. I'm not sure what's the smartest thing to do.
The issue with top-level is that it pollutes a bit and really few or no one is going to use the frontend in a standalone expr.

Whereas having a proper package set to enclose the namespace makes it still easy to override the drv properly.

Otherwise, a technique I have been using a lot is to have a second level of function:
```
{ cmake, nixosTests, ... }:
let mkMobilizonFinalPackage = { frontend, ... }: mkDerivation { ... }; in (mkMobilizonFinalPackage { defaults }) // { passthru = { inherit mkMobilizonFinalPackage; }; }
```

I do not recommend it as it has very poor semantics wrt to override/etc.

Reply:
`mobilizon.frontend` sounds the most sensible to me. Could you see yourself implementing that? Because I'm a bit unsure on how to achieve that.

Reply:
I can take a look if you offer me the commit bits on the original branch, cc @minijackson

Reply:
@RaitoBezarius I'm sorry but I'd really prefer not giving write access to my fork to too many people (especially since I don't know if it's possible to restrict branches to certain collaborators)

Reply:
No problem, @minijackson
I'd advise poking around and look for other frontend/backend projects which might have some `XXXFrontend` package in the head function.

Reply:
@SuperSandro2000 `finalAttrs` looks really cool, never heard of it before! I think I'd have to change the definition of `mixRelease` to allow a function as argument. Are you aware of any comparable efforts I could look at? A quick grep didn't lead me to any.

Reply:
I removed tha `srcOverride` argument here and chose to go for an approach similar to the one I found for the `woodpecker` package.